### ISSUE TITLE
UNO IP Arm Home or Night through Home Assistant sends Arm Away to board

### ISSUE THREAD

**Issue by copperlab** [ufodone/envisalink_new#147]
When I issue an arm home or arm night command from HA it always actually goes into home away on the UNO IP system.

I am not sure if this is one of the features that has not been added yet.  Other than that seems to be working great.

**Comment by ufodone** [comment]
Yes, this was a current limitation of the UNO's TPI interface when I originally added support for it.  I've not seen any firmware release notes indicating updates yet but I've sent a note off to enquire if/when additional features are coming.

**Comment by copperlab** [comment]
I reached out to eyezon about tpi interface.  They said that is controlled by the manufacturer which is Envisacor.  To me it looks like they are the same company.  I can help ask for whatever you need, but I just don't know exactly what would be helpful for you.  Thanks.

**Comment by ufodone** [comment]
Yes, it is Envisacor I've contacted.  I know they had plans to add a bunch more functionality but don't know a timeline on it at the moment.

**Comment by copperlab** [comment]
I also reached out on the eyezon forum about new TPI documentation.   The support person I talked to at eyezon says they do answer questions on their forum.  Let me know if you get a response.  Thanks.

**Comment by ufodone** [comment]
They are just working on updated documentation and will send me a copy when it is complete.  It sounds like that will be relatively soon.  This should capture additional capabilities they are adding.  Based on previous conversations I've had with them, I suspect these updates will have the ability to arm in different modes.

**Comment by deliverer33** [comment]
not sure if it helps the cause or not, but you can arm "home" from the eyezon web interface.  does that help to gather what is needed to update the code?

**Comment by copperlab** [comment]
any updates on this?

**Comment by ufodone** [comment]
@copperlab No, I haven't heard anything back from Envisacor yet.  

@deliverer33 Unfortunately the interface used by this integration does not have the same capabilities as what has been made available on the web interface.  So it's just a matter of waiting until they release firmware and documentation with the additional capabilities.

**Comment by copperlab** [comment]
I see a new firmware was released.  Did this come with what you were looking for?

**Comment by tamorgen** [comment]
Any updates?  I'm unable to arm in night or home mode through HAAS on my UNO IP hybrid system.

**Comment by copperlab** [comment]
There is another new update that was released today.  Also have a thread on the eyezon forum to try to get the attention of the manufacturer: https://forum.eyezon.com/viewtopic.php?t=5442&sid=11530eabfa0b5339462928e48ea92731

**Comment by tamorgen** [comment]
> There is another new update that was released today. Also have a thread on the eyezon forum to try to get the attention of the manufacturer: https://forum.eyezon.com/viewtopic.php?t=5442&sid=11530eabfa0b5339462928e48ea92731

I just updated to 117 this morning, and it still tries to arm away when trying night or home

**Comment by ufodone** [comment]
I spoke with Envisacor and the released firmware versions do not yet include the TPI changes needed.  While I don't have an ETA, they are actively work on the required changes.

**Comment by tamorgen** [comment]
> I spoke with Envisacor and the released firmware versions do not yet include the TPI changes needed. While I don't have an ETA, they are actively work on the required changes.

Well, that is at least something.  They are acknowledging there is an issue.  The firmware updates seem to be somewhat regular, so hopefully they'll get this sorted out.  Until then, I'll need to arm my system manually.

**Comment by ufodone** [comment]
> Well, that is at least something. They are acknowledging there is an issue.

They've actually been very responsive with me and very receptive to adding features to the TPI.  Development sometimes just takes time and I'm sure they have many competing priorities to juggle.

**Comment by tamorgen** [comment]
> > Well, that is at least something. They are acknowledging there is an issue.
> 
> They've actually been very responsive with me and very receptive to adding features to the TPI. Development sometimes just takes time and I'm sure they have many competing priorities to juggle.

Okay, that's good to hear.  I guess reading the entire thread, it didn't seem like they were entirely responsive at first.  Then again, that was 2 months ago with a limited number of responses.

**Comment by DineshChr** [comment]
I updated my Uno panel with the recent beta version and now Arm_stay (home) works. I updated home assistant with the current beta version, restarted HA and Waited for about 6 Hrs and now both Arm_away and Arm_stay functions are working in Apple HomeKit.

**Comment by ufodone** [comment]
I just heard back from Envisacor and they have a new firmware available which adds additional capabilities to the TPI interface.  

I still need to make the necessary changes to support it in this integration but I don't own an Uno version so will need some people to test it.

Are any of you willing to have the new firmware pushed to your devices to help test?  If so, please send me the MAC address of your device and I can pass that along to Envisacor.  Thanks!

**Comment by deliverer33** [comment]
I am 

001C2A400364

Current Firmware - Firmware: 01.00.104

**Comment by DineshChr** [comment]
Mine:
001C2A400DE3


From: Will Butler ***@***.***>
Date: Monday, January 27, 2025 at 10:11 PM
To: ufodone/envisalink_new ***@***.***>
Cc: Dinesh Christendat ***@***.***>, Comment ***@***.***>
Subject: Re: [ufodone/envisalink_new] UNO IP Arm Home or Night through Home Assistant sends Arm Away to board (Issue #147)
You don't often get email from ***@***.*** Learn why this is important<https://aka.ms/LearnAboutSenderIdentification>

I am

001C2A400364

Current Firmware - Firmware: 01.00.104

**Comment by tamorgen** [comment]
> I just heard back from Envisacor and they have a new firmware available which adds additional capabilities to the TPI interface.
> 
> I still need to make the necessary changes to support it in this integration but I don't own an Uno version so will need some people to test it.
> 
> Are any of you willing to have the new firmware pushed to your devices to help test? If so, please send me the MAC address of your device and I can pass that along to Envisacor. Thanks!

I'm in.

001C2A4013A6

Current Firmware:  117, although it is telling me 122 is available.   Queueing that up now.

**Comment by deliverer33** [comment]
> I am
> 
> 001C2A400364
> 
> Current Firmware - Firmware: 01.00.104

Saw new firmware 122 so updated as of this morning.

**Comment by ufodone** [comment]
The firmware version with the necessary changes is `1.0.123`.  If you got your Uno through EyezOn, I'm told you can ask them to update your device to this firmware now.  If you are  Connect2Go customer, it sounds like I'll need to ask Envisacor to push the update to you.

Before anyone does this, a couple things:
1. Because this is an early/initial release, there may be bugs and other issues that may be caused either by issues in this integration or in issues with the new firmware (or a combination of both).  To that end, I ask that you only sign up for this if you're okay with the possibility of issues cropping up and that it may take some time to resolve them.  My hope is that it will all go relatively smoothly but there is always a chance off issues and I don't want anyone stuck if this integration is a "mission critical" part of their setup.
2. I am working on the necessary changes to this integration to support the additional features.  I'm told the new firmware should be backward compatible so theoretically it should be okay to update before I have a new version of this integration available.  But I would suggest that you wait to update the firmware until I have a beta version ready.  I'm working away on it now but it will likely be at least a few more days before I have something ready (it's a bit slow going without an Uno device to test against).  

I'll update here when I have a beta version available.

**Comment by deliverer33** [comment]
> The firmware version with the necessary changes is `1.0.123`. If you got your Uno through EyezOn, I'm told you can ask them to update your device to this firmware now. If you are Connect2Go customer, it sounds like I'll need to ask Envisacor to push the update to you.
> 
> Before anyone does this, a couple things:
> 
> 1. Because this is an early/initial release, there may be bugs and other issues that may be caused either by issues in this integration or in issues with the new firmware (or a combination of both).  To that end, I ask that you only sign up for this if you're okay with the possibility of issues cropping up and that it may take some time to resolve them.  My hope is that it will all go relatively smoothly but there is always a chance off issues and I don't want anyone stuck if this integration is a "mission critical" part of their setup.
> 2. I am working on the necessary changes to this integration to support the additional features.  I'm told the new firmware should be backward compatible so theoretically it should be okay to update before I have a new version of this integration available.  But I would suggest that you wait to update the firmware until I have a beta version ready.  I'm working away on it now but it will likely be at least a few more days before I have something ready (it's a bit slow going without an Uno device to test against).
> 
> I'll update here when I have a beta version available.

I've got the updated code, 1.0.123 pushed to my device.

**Comment by andredigenova** [comment]
Looks like version 123 is live to everyone now.

**Comment by ufodone** [comment]
I have published a [beta release](https://github.com/ufodone/envisalink_new/releases/tag/v1.1.0-uno-beta1) based on the changes in the latest firmware.

Even if you don't find any issues, I would appreciate someone posting some debug logs from startup and when issuing various commands (arm, disarm, zone bypass, etc.) so I have get a better idea of the traffic to/from the Uno device.

There is still more to add but I wanted to get something out for people to try.

**Comment by DineshChr** [comment]
> Looks like version 123 is live to everyone now.

I am on firmware version 122
How do I update to 123?

**Comment by tamorgen** [comment]
> > Looks like version 123 is live to everyone now.
> 
> I am on firmware version 122 How do I update to 123?

I'm in the same situation.  I'm on 122, but I'm not being prompted for 123.  I'm not sure what to do to get it to provide the update.

**Comment by deliverer33** [comment]
I’m on the new beta and can confirm Arm Home works as does Arm Away.  One interesting item I cannot disarm until the alarming period completes.  So I can’t click Arm Home and then disarm right away.  Happy to send logs just let me know how and what.

Great work and thank you!

**Comment by andredigenova** [comment]
> > > Looks like version 123 is live to everyone now.
> > 
> > 
> > I am on firmware version 122 How do I update to 123?
> 
> I'm in the same situation. I'm on 122, but I'm not being prompted for 123. I'm not sure what to do to get it to provide the update.

maybe a phased rollout? I am not on any beta program and was prompted for version 123 after being on version 117

**Comment by DineshChr** [comment]
> > > > Looks like version 123 is live to everyone now.
> > > 
> > > 
> > > I am on firmware version 122 How do I update to 123?
> > 
> > 
> > I'm in the same situation. I'm on 122, but I'm not being prompted for 123. I'm not sure what to do to get it to provide the update.
> 
> maybe a phased rollout? I am not on any beta program and was prompted for version 123 after being on version 117

Would it be fine to install the new beta version with firmware 122?

**Comment by ufodone** [comment]
I suspect they do a phased rollout.  If your device is from EyezOn, I believe you should be able to contact them and request that they push the new firmware to your device.

> Would it be fine to install the new beta version with firmware 122?

@DineshChr No, the beta will only work with the 123 firmware as it relies on new features added to that version.

**Comment by ufodone** [comment]
You can install it via HACS.  Click on the HACS link on the left side of your HA screen, find this integration, click the overflow menu (far right) and choose "Redownload" and you should be able to find the beta version in the list of versions there.

The beta depends on the new firmware and will likely render the integration unusable with an older firmware, I'm waiting for a bit to let the new firmware propagate out to devices before I promote this beta to an official release.

**Comment by tamorgen** [comment]
> You can install it via HACS. Click on the HACS link on the left side of your HA screen, find this integration, click the overflow menu (far right) and choose "Redownload" and you should be able to find the beta version in the list of versions there.
> 
> The beta depends on the new firmware and will likely render the integration unusable with an older firmware, I'm waiting for a bit to let the new firmware propagate out to devices before I promote this beta to an official release.

Thanks.  I forgot you can load versions through HACS.  It updated just fine.

The system armed just fine the first time when I was away, but I noticed it wasn't arming at night.  It looks like the Arm Night was removed from this release of the firmware.  Honestly, I'm not sure what the difference between Arm Night Arm and  Arm Home is anyway.  I changed my automation to use Arm Home and it appears to be working just fine.

I did notice on the EyezOn portal, there is an Arm Home, Away Arm Home, and Stay Arm Home.  Maybe Stay Arm Home is what Night Arm used to be?  Again, I'm not sure the difference between the Arm Home and Stay Arm Home.
